src/power_grid_model/validation/rules.py

```diff
@@ -754,6 +754,53 @@ def all_finite(data: SingleDataset, exceptions: dict[ComponentType, list[str]] |
     return errors
 
 
+def subset_of_fields_are_missing(data: SingleDataset, fields : list[str], component_type : ComponentType):
+    """
+    Helper function that generates missing value errors if a subset of the supplied fields are missing.
+    If for an instance of component type all fields are missing or all fields are not missing then no error is returned for that instance.
+    In any other case an error for that id is returned.
+
+    Args:
+        data: SingleDataset, pgm data
+        fields: List of fields
+        component_type: component type to check
+    """
+    errors = []
+    if component_type in data:
+        asym_line_data = data[component_type]
```

Comment:
the rules in `rules.py` should be more generic. please use a more generic name
```suggestion
        component_data = data[component_type]
```

Reply:
check